packages/app/src/components/search/SearchPage.tsx

```diff
@@ -55,6 +55,8 @@ const SearchPage = () => {
   const classes = useStyles();
   const { isMobile } = useContext(SidebarPinStateContext);
 
+  useInfiniteScrollDocumentSearch();
```

Comment:
(Heh, whoops.  Per my comment above, I'd encourage y'all to steer toward the latter, although packaging up the infinite scroll option in a way where hook knowledge is also not required would be ideal too!)

Reply:
yes, this is a grey area.
In the case of the Modal, there is a specific scrollable element. As opposed to the Modal, in the SearchPage there is none: the scrollable element is `<html>`. For this reason I had to create a different hook covering this specific use case. I’m not sure I could create an universal approach for solving both problems in one shot, as this totally depends on where in the page the elements are rendered.

Regarding the proposal to “hide” the hook. This could be solved using an approach like:

```javascript
<SearchContainer>
    <Container>
     ….
         <SearchResult>
         {(results => … )}
         </SearchResult>
     ….
    </Container>
```

Where SearchContainer will add the scroll listener to the child component.

My question is: are we sure this is an easier approach? From an adopter perspective, the constraint is that SearchContainer needs to be positioned immediately above the scrollable container. What if the user misses this?

From an implementation perspective, building something like `SearchContainer` is a bit more complex, since internally it needs to inject some props to the children.

These are the main two reasons why I’ve chosen the hook approach.

On the other side a good point about the `SearchContainer` is that the two use cases mentioned above (SearchPage, SearchModal) can be hidden internally in the component (maybe configurable using a boolean prop).

Thoughts?

Reply:
Hmm...  What about something like...  Keep `<SearchResultPager />` around, but add a `<SearchResultPager infinite />` prop?  When `infinite` is set/true, then just automatically trigger another page of results if the `<SearchResultPager />` instance is visible on screen?

Reply:
how could `SearchResultPager ` know which the "scrollable" element is in the rendered page?

Reply:
yeah I like it!

Reply:
Was thinking about something like this: https://usehooks.com/useOnScreen/

Reply:
sounds good!

Reply:
so, now the problem of using `useOnScreen` is the following (apart from some fixes needed in the example):

since the results and `SearchResultPager` are in two different places, `SearchResultPager` can be visible before all `SearchResult` are rendered on the screen, triggering an unexpected fetch. 😩

Reply:
Hmmmm...  Any way to combine it with the `useSearch()` context so that it only fetches if results are not currently loading?

Reply:
yeah the problem is that the rendering of the results happen asynchronously, after the results have been stored in the context. So combining `SearchResultPager` with the results coming from the context didn't help.

BTW this could be another approach that doesn't introduce external hooks: https://github.com/backstage/backstage/compare/vinzscam/search-infinite-scrolling...vinzscam/search-infinite-scrolling-without-external-hooks?expand=1